platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/fingerprint/ConfigurationCacheFingerprintWriter.kt

```diff
@@ -923,6 +932,97 @@ class ConfigurationCacheFingerprintWriter(
     override fun onScriptFileResolved(scriptFile: File) {
         fileObserved(scriptFile)
     }
+
+    override fun onGradlePropertiesByPrefix(
+        propertyScope: GradlePropertiesListener.PropertyScope,
+        prefix: String,
+        snapshot: Map<String, String>
+    ) {
+        if (isInputTrackingDisabled()) {
+            return
+        }
+        if (gradlePropertiesByPrefix.computeIfAbsent(propertyScope) { newConcurrentHashSet() }.add(prefix)) {
+            sink().write(
+                ConfigurationCacheFingerprint.GradlePropertiesPrefixedBy(
+                    propertyScope,
+                    prefix,
+                    snapshot
+                )
+            )
+            reportGradlePropertiesByPrefixInput(propertyScope, prefix)
+        }
+    }
+
+    override fun onGradlePropertyAccess(
+        propertyScope: GradlePropertiesListener.PropertyScope,
+        propertyName: String,
+        propertyValue: Any?
+    ) {
+        if (isInputTrackingDisabled()) {
+            return
+        }
+        if (gradleProperties.computeIfAbsent(propertyScope) { newConcurrentHashSet() }.add(propertyName)) {
+            sink().write(
+                ConfigurationCacheFingerprint.GradleProperty(
+                    propertyScope,
+                    propertyName,
+                    propertyValue
+                )
+            )
+            reportGradlePropertyInput(propertyScope, propertyName)
+        }
+    }
+
+    private
+    fun reportGradlePropertyInput(
+        propertyScope: GradlePropertiesListener.PropertyScope,
+        propertyName: String,
+        consumer: String? = null
+    ) {
+        if (propertyName.startsWith("org.gradle.")) {
+            // Don't include builtin Gradle properties in the report
```

Comment:
💭  They may not be in the report, but can they still be the reason for a cache miss as printed on the console?